tools/mason/MasonSearch.chpl

```diff
@@ -122,6 +119,23 @@ proc masonSearch(ref args: list(string)) {
   }
 }
 
+
+/* Sort the results in a order such that results that startWith needle
+   are displayed first */
+proc rankResults(results, query) {
```

Comment:
I think we should continue to sort alphabetically if there is no match, until a more advanced score-based sorting is available.

I think the error I am seeing on my machine is due to OS X `listdir` listing the files in a different order. Since those files are never sorted beyond the comparator below, they are out of order for the `no-pattern.good` test on OS X.

Reply:
@ankingcodes - the default `mason search` now gives me the same thing (🎉 !). However, `mason-search.chpl` still fails for me on `no-pattern.good` with the output:

```
WithCaps (0.1.0)
capsize (0.1.0)
multiVersion (1.0.0)
```

instead of:

```
multiVersion (1.0.0)
WithCaps (0.1.0)
capsize (0.1.0)
```